doc/developers/maintainer.rst

```diff
@@ -230,50 +224,56 @@ Making a release
      git tag -a 0.99.0  # in the 0.99.X branch
      git push git@github.com:scikit-learn/scikit-learn.git 0.99.0
 
-6. Trigger the GitHub Actions workflow again but this time to upload the artifacts
+6. Confirm that the bot has detected the tag on the conda-forge feedstock repo:
+   https://github.com/conda-forge/scikit-learn-feedstock. If not, submit a PR for the
+   release. If you want to publish an RC release on conda-forge, the PR should target
+   the `rc` branch as opposed to the `main` branch. The two branches need to be kept
+   sync together otherwise.
+
+7. Trigger the GitHub Actions workflow again but this time to upload the artifacts
    to the real https://pypi.org (replace "testpypi" by "pypi" in the "Run
    workflow" form).
 
-7. Alternatively, it's possible to collect locally the generated binary wheel
-   packages and source tarball and upload them all to PyPI by running the
-   following commands in the scikit-learn source folder (checked out at the
-   release tag):
+   7.1. Alternatively, it's possible to collect locally the generated binary wheel
```

Comment:
The rendering is a [bit weird](https://output.circle-artifacts.com/output/job/679d538a-a0b2-4f3e-a4ea-c8729d5e6a33/artifacts/0/doc/developers/maintainer.html). I think the simplest solution is to make 7.1 -> 8. and then have in bold:

```rst
8. **Alternative to Step 7**: It's possible to collect locally ...
```

Reply:
there was already the same issue with step 4.1. Your proposition should make it cleaner.